### PR TITLE
[Fix] Update links in readme of DETRs and MaskFormers

### PR DESCRIPTION
**Pull request by Li-Qingyun**
We (me, @jshilong, @LYMDLUT, and @KeiChiTse) have refactored DETR-like models, to enhance the usability and readability of our codebase. This PR update links in readme of DETRs and MaskFormers.

### CODE REVIEW

File: configs/deformable_detr/README.md [hunk 1]
```diff
@@ -14,11 +14,11 @@ DETR has been recently proposed to eliminate the need for many hand-designed com
...
+| :------: | :---------------------------------: | :-----: | :----: | :---------------------------------------------------------------: | :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
+|   R-50   |           Deformable DETR           |   50e   |  44.5  |         [config](./deformable-detr_r50_16xb2-50e_coco.py)         |                            [model](https://download.openmmlab.com/mmdetection/v3.0/deformable_detr/deformable-detr_r50_16xb2-50e_coco/deformable-detr_r50_16xb2-50e_coco_20221029_210934-6bc7d21b.pth) \| [log](https://download.openmmlab.com/mmdetection/v3.0/deformable_detr/deformable-detr_r50_16xb2-50e_coco/deformable-detr_r50_16xb2-50e_coco_20221029_210934.log.json)                             |
```
Comment (jshilong) [inline configs/deformable_detr/README.md:19]: fix the performacne 44.5 ?

File: configs/dino/README.md [hunk 2]
```diff
@@ -14,9 +14,9 @@ We present DINO (DETR with Improved deNoising anchOr boxes), a state-of-the-art
...
+| Backbone | Model | Lr schd | box AP |                    Config                    |                                                                                                                                                  Download                                                                                                                                                  |
+| :------: | :---: | :-----: | :----: | :------------------------------------------: | :--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
+|   R-50   | DINO  |   12e   |  49.0  | [config](./dino_4scale_r50_8xb2-12e_coco.py) | [model](https://download.openmmlab.com/mmdetection/v3.0/dino/dino_4scale_r50_8xb2-12e_coco/dino_4scale_r50_8xb2-12e_coco_20221202_182705-55b2bba2.pth) \| [log](https://download.openmmlab.com/mmdetection/v3.0/dino/dino_4scale_r50_8xb2-12e_coco/dino_4scale_r50_8xb2-12e_coco_20221202_182705.log.json) |
```
Comment (jshilong) [inline configs/dino/README.md:19]: Add a comment to explain this performance is  fluctuating
and add comments for the normal range.

File: configs/conditional_detr/README.md [hunk 3]
```diff
@@ -23,9 +23,9 @@ Our conditional DETR learns a conditional spatial query from the decoder embeddi
...
+| :------: | :--------------: | :-----: | :------: | :------------: | :----: | :-----------------------------------------------: | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
+|   R-50   | Conditional DETR |   50e   |   7.9    |                |  40.9  | [config](./conditional_detr_r50_8xb2-50e_coco.py) | [model](https://download.openmmlab.com/mmdetection/v3.0/conditional_detr/conditional_detr_r50_8xb2-50e_coco/conditional_detr_r50_8xb2-50e_coco_20221121_180202-c83a1dc0.pth) \| [log](https://download.openmmlab.com/mmdetection/v3.0/conditional_detr/conditional_detr_r50_8xb2-50e_coco/conditional_detr_r50_8xb2-50e_coco_20221121_180202.log.json) |
```
Comment (ZwwWayne) [inline configs/conditional_detr/README.md:28]: conditional_detr -> conditional-detr

File: configs/dino/README.md [hunk 4]
```diff
@@ -14,9 +14,13 @@ We present DINO (DETR with Improved deNoising anchOr boxes), a state-of-the-art
...
+| Backbone |    Model    | Lr schd | box AP |                    Config                    |                                                                                                                                                  Download                                                                                                                                                  |
+| :------: | :---------: | :-----: | :----: | :------------------------------------------: | :--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
+|   R-50   | DINO-4scale |   12e   |  49.0  | [config](./dino_4scale_r50_8xb2-12e_coco.py) | [model](https://download.openmmlab.com/mmdetection/v3.0/dino/dino_4scale_r50_8xb2-12e_coco/dino_4scale_r50_8xb2-12e_coco_20221202_182705-55b2bba2.pth) \| [log](https://download.openmmlab.com/mmdetection/v3.0/dino/dino_4scale_r50_8xb2-12e_coco/dino_4scale_r50_8xb2-12e_coco_20221202_182705.log.json) |
```
Comment (ZwwWayne) [inline configs/dino/README.md:19]: dino_4scale -> dino-4scale

**Comment by ZwwWayne** [comment]
LGTM after fixing naming issue

**Comment by Li-Qingyun** [comment]
> LGTM after fixing naming issue

The renaming has been finished ~

**Comment by Li-Qingyun** [comment]
The Unit tests passes on my laptop

File: configs/maskformer/README.md [hunk 5]
```diff
@@ -36,10 +36,15 @@ mmdetection
...
+| :------: | :-----: | :-----: | :------: | :------------: | :----: | :----: | :----: | :----: | :----: | :----: | :----: | :----: | :----: | :--------------------------------------------------------: | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
+|   R-50   | pytorch |   75e   |   16.2   |       -        | 46.757 | 80.297 | 57.176 | 50.829 | 81.125 | 61.798 | 40.610 | 79.048 | 50.199 |      [config](./maskformer_r50_ms-16xb1-75e_coco.py)       |                           [model](https://download.openmmlab.com/mmdetection/v3.0/maskformer/maskformer_r50_ms-16xb1-75e_coco/maskformer_r50_ms-16xb1-75e_coco_20230116_095226-baacd858.pth) \| [log](https://download.openmmlab.com/mmdetection/v3.0/maskformer/maskformer_r50_ms-16xb1-75e_coco/maskformer_r50_ms-16xb1-75e_coco_20230116_095226.log.json)                           |
```
Comment (jshilong) [inline configs/maskformer/README.md:41]: add a comment here to indicate this performance is fluctuating and it is a slightly lower than original paper

**Review by jshilong — APPROVED** [review]
This PR can be merged after @Li-Qingyun  check all the link

**Pull request merged**